src/blockdata/script.rs

```diff
@@ -501,12 +507,17 @@ impl Script {
         Ok(())
     }
 
-    /// Get the assembly decoding of the script.
-    pub fn asm(&self) -> String {
+    /// Create an assembly decoding of the script in the given byte slice.
+    pub fn str_asm(script: &[u8]) -> String {
```

Comment:
Hmm yeah naming. I figured not too important is't not really gonna be a high-usage method. `asm` is a member method, that would be used more. I made `str_asm` kinda cfr `fmt_asm`, but `asm_from_bytes` makes sense.

Reply:
I think the good naming for the associated conversion function should include both original and return data type, like `bytes_to_string` (`str` is reserved for `&str` return type). Thus, it will be `Script::bytes_to_string`, which is very readable and explicit.

Reply:
I disagree on the return type being in the name, `string` is as non-descriptive as it gets (could be hex for all we know), this should remain the chosen encoding, i.e. `asm`. I'd be ok with `bytes_to_asm`, but don't really care as long as it's descriptive.

Reply:
Agree on `bytes_to_asm`

Reply:
@stevenroose Would you be ok with giving this a more descriptive name? Rarely used or not, it is part of the public API we will be stuck with for a while.